### PR TITLE
Support :straight (pkg-a pkg-b) syntax in use-package

### PR DESCRIPTION
**Pull request by justbur**
Previously this resulted in an error because it expanded
to (straight-use-package '(pkg-a pkg-b)).

### CODE REVIEW

**Comment by raxod502** [comment]
What you want is already supported:

```elisp
(use-package pkg
  :straight pkg-a
  :straight pkg-b)
```

Also keep in mind that the following syntax must also be supported:

```elisp
(use-package tex
  :recipe (auctex :type git :host github :repo "emacsmirror/auctex"))
```

**Comment by justbur** [comment]
The syntax `:straight (pkg-a pkg-b)` is common in `use-package` (see `:commands`, `:load-path`, and more), is not treated as an error, and gives an invalid expansion. 

You are supporting the example you gave, as well as `:straight pkg-a pkg-b`, but both of those seem unconventional compared to `:straight (pkg-a pkg-b)`. 

I don't understand the significance of your second point, unless you meant to write `:straight (auctex :type git :host github :repo "emacsmirror/auctex")`.

**Comment by justbur** [comment]
Just looked at the README again. I see the problem. Let me try something else

**Comment by justbur** [comment]
@raxod502 Can you see if it's acceptable now? I detect recipes by the presence of a keyword and otherwise recursively process the arg.

File: straight.el [hunk 1]
```diff
@@ -4586,10 +4586,11 @@ documentation."
        ((null arg) (setq parsed-args nil))
        ((eq arg t) (push name-symbol parsed-args))
        ((symbolp arg) (push arg parsed-args))
-       ((not (listp arg))
-        (use-package-error ":straight wants a symbol or list"))
```
Comment (raxod502) [inline straight.el]: Why did you move this condition down to the end? It seems like that introduces a duplicate check for `(listp arg)`, instead of having just one.

**Comment by raxod502** [comment]
Regarding the substance of the change, I'm not opposed to allowing multiple packages and/or recipes wrapped in a list rather than as separate top-level forms.

But we also need to continue supporting a single recipe at the top-level, which your change doesn't currently do. In addition to checking the first element of a list, you need to check the _second_ element to see if it's a keyword—that would indicate a recipe, like in:

```elisp
(use-package tex
  :straight (auctex :type git :host github :repo "emacsmirror/auctex"))
```

rather than a list of package names `(pkg-a pkg-b)`.

Also, for consistency, I'd like to support lists of recipes and/or package names:

```elisp
(use-package tex
  :straight (no-littering (auctex :type git :host github :repo "emacsmirror/auctex")))
```

which would translate to

```elisp
(straight-use-package 'no-littering)
(straight-use-package '(auctex :type git :host github :repo "emacsmirror/auctex"))
```

File: straight.el (hunk 1 above)
Comment (raxod502) [reply straight.el]: This might be outdated though, it seems that the branch has been deleted.

**Comment by raxod502** [comment]
Oh, I see this was moved to a new pull request. Just FYI, GitHub does let you change the base branch of a pull request and also force-push new commits to the target branch, so you don't have to create a new pull request for every revision.
